src/test/resources/projects/it1/reference/nexus-repository-foo-it/src/test/java/org/sonatype/nexus/plugins/foo/internal/FooITSupport.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) ${ITReferenceYear}-present Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.plugins.foo.internal;
+
+import java.net.URL;
+
+import javax.annotation.Nonnull;
+
+import org.sonatype.nexus.pax.exam.NexusPaxExamSupport;
+import org.sonatype.nexus.plugins.foo.internal.fixtures.RepositoryRuleFoo;
+import org.sonatype.nexus.repository.Repository;
+import org.sonatype.nexus.testsuite.testsupport.RepositoryITSupport;
+
+import org.junit.Rule;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class FooITSupport
+    extends RepositoryITSupport
+{
+  @Rule
+  public RepositoryRuleFoo repos = new RepositoryRuleFoo(() -> repositoryManager);
+
+  @Override
+  protected RepositoryRuleFoo createRepositoryRule() {
+    return new RepositoryRuleFoo(() -> repositoryManager);
+  }
+
+  public FooITSupport() {
+    testData.addDirectory(NexusPaxExamSupport.resolveBaseFile("target/it-resources/foo"));
```

Comment:
maybe also here we can make the `foo` reference repository name based or at least generic (`repodata`)

Reply:
These are "archetype IT reference" classes: https://maven.apache.org/archetype/maven-archetype-plugin/integration-test-mojo.html
The exist to ensure the generated code is as expected.